cns/ipampool/monitor.go

```diff
@@ -123,9 +124,17 @@ func (pm *Monitor) Start(ctx context.Context) error {
 			// Add Primary IP to Map, if not present.
 			// This is only for Swift i.e. if NC Type is vnet.
 			for i := 0; i < len(nnc.Status.NetworkContainers); i++ {
-				if nnc.Status.NetworkContainers[i].Type == "" ||
-					nnc.Status.NetworkContainers[i].Type == v1alpha.VNET {
-					pm.metastate.primaryIPAddresses[nnc.Status.NetworkContainers[i].PrimaryIP] = struct{}{}
+				nc := nnc.Status.NetworkContainers[i]
+				if nc.Type == "" || nc.Type == v1alpha.VNET {
+					pm.metastate.primaryIPAddresses[nc.PrimaryIP] = struct{}{}
+				}
+
+				if nc.Type == v1alpha.VNETBlock {
+					primaryPrefix, err := netip.ParsePrefix(nc.PrimaryIP)
+					if err != nil {
+						return errors.Wrapf(err, "IP: %s", nc.PrimaryIP)
+					}
+					pm.metastate.primaryIPAddresses[primaryPrefix.Addr().String()] = struct{}{}
```

Comment:
We use `.Masked()` in the conversion.go to get the 0th address, should we do that here also? If we get `192.168.0.5/28`, is the primary the `.5` address, or the `.0` address?

Reply:
In that case, i believe primary ip is .5 I guess shouldn't have used masked in conversion.go. 
Updating

Reply:
would we ever get 192.168.0.5/28? That is an invalid CIDR, I would expect it to fail parsing before this code block

Reply:
I am not enforcing that validation in CNS. I was assuming DNC provides a valid primaryIP CIDR. FYI @nairashu

Reply:
DNC should be passing down correct CIDR ranges. Maybe we should error out a metric if the range is invalid

Reply:
`192.168.0.5/28` IS valid CIDR notation, and it parses fine.
The question is, IF we get such an address, do we take *that* address (`192.168.0.5`) to be the NC Primary IP, or do we take the network address of the block (the 0th address, `192.168.0.0`)?
In Overlay, we have taken the 0th address of the block, but now I'm wondering if that's guaranteed.

Reply:
So the IP defined in the primary IP field will be taken by PubSub as the Primary IP address by removing the / from the CIDR notation. We will have to also adhere to that contract. That being said we should only be passing down an address where the /27 CIDR is the 0th address of the CIDR block.

Reply:
We should have validations in DNC code for this purpose during CIDR allocations to the NC.

Reply:
@thatmattlong is this true for Overlay also?

Reply:
Synced with Ashish. Seems like I need to use Primary address as is(without masking) since in NMAgent the programming will be to remove the / from the IP mentioned for Primary when NC is created. We will however be passing down address where /27 CIDR is the 0th address of the block.  The validation will be in DNC though.

@rbtr Pls resolve this if your concern has been addressed. :)